Во время нашего путешествия обучения Python мы сталкивались с ошибками, сгенерированными интерпретатором кучу раз. Мы видели примеры TypeError, IndexError, ValueError и других. До сих пор всякий раз, когда интерпретатор бросал одну из этих ошибок, мы меняли наш код, чтобы избежать ошибки. Это общий подход, так как всякий раз, когда интерпретатор поднимает одну из этих ошибок, программа останавливается, и мы не хотим, чтобы наши сценарии заканчивались до того, как они закончат свою работу. Иногда проще сделать верификацию с условным условием, чтобы избежать ошибки. Как и в нашем предыдущем примере измененной функции имени, где мы проверяем, не был ли результат нашего поиска в регулярном выражении и сделал что-то другое в этом случае. В другие времена есть так много вещей, которые могут пойти не так, что проверка всех из них становится сложной задачей. 